prompt2model/param_selector/search_with_optuna.py

```diff
@@ -0,0 +1,188 @@
+            "max_weight_decay": 1e-1,
+            "min_learning_rate": 1e-5,
+            "max_learning_rate", 1e-1,
+        }
+        ```
+        Returns:
+            Returns the transformers.Trainer class with the optimal hyperparameters
+        """
+        # TODO:
+        # - Find the industry standards for default values spec
+        # - More asserts for other keys. Example checking the min or max values
+
+        if "save_strategy" in hyperparameters:
+            save_strategy_is_valid = any(
+                strategy in ["epoch", "steps", "no"]
+                for strategy in hyperparameters["save_strategy"]
+            )
+            assert (
+                save_strategy_is_valid
+            ), "save strategy should have either of these values: ['epoch', 'steps', 'no']"
+
+        if "evaluation_strategy" in hyperparameters:
+            evaluation_strategy_is_valid = any(
+                strategy in ["epoch", "no"]
+                for strategy in hyperparameters["evaluation_strategy"]
+            )
+            assert (
+                evaluation_strategy_is_valid
+            ), "Evaluation strategy should have either of these values ['epoch', 'no']"
+
+        def hp_space(trial: Trial):
+            return {
+                "num_train_epochs": trial.suggest_int(
+                    "num_train_epochs",
+                    hyperparameters.get("min_train_epochs", 5),
+                    hyperparameters.get("max_train_epochs", 10),
+                ),
+                "save_strategy": trial.suggest_categorical(
+                    "save_strategy",
+                    hyperparameters.get("save_strategy", ["epoch", "steps", "no"]),
+                ),
+                "evaluation_strategy": trial.suggest_categorical(
+                    "evaluation_strategy",
+                    hyperparameters.get("evaluation_strategy", ["epoch", "no"]),
+                ),
+                "per_device_train_batch_size": trial.suggest_categorical(
+                    "per_device_train_batch_size",
+                    hyperparameters.get("per_device_train_batch_size", [4, 8, 16, 32]),
+                ),
+                "weight_decay": trial.suggest_loguniform(
+                    "weight_decay",
+                    hyperparameters.get("min_weight_decay", 1e-5),
+                    hyperparameters.get("max_weight_decay", 1e-1),
+                ),
+                "learning_rate": trial.suggest_loguniform(
+                    "learning_rate",
+                    hyperparameters.get("min_learning_rate", 1e-5),
+                    hyperparameters.get("max_learning_rate", 1e-2),
+                ),
+            }
+
+        # prepare the training args
+        # we are assuming here that the user will also provide these args with the additional
+        # args for range. Or we can provide an another argument of search_args (dict) that will
+        # tell the user to provide the arguments for doing hyperparamerter search
```

Comment:
Comments should almost always be complete, grammatical sentences (with proper capitalization and punctuation). I also feel that most of this comment is unnecessarily detailed/confusing.

Reply:
Yeah, currently I am struggling on this part, but will improve on that